test/lib/y2remote/display_manager_test.rb

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env rspec
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LLC
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require_relative "../../test_helper.rb"
+require "y2remote/remote"
+
+describe Y2Remote::DisplayManager do
+  subject { described_class.instance }
+
+  describe ".enabled?" do
+    let(:service_enabled) { false }
+
+    before do
+      stub_const("Yast::Service", double(Enabled: service_enabled))
+    end
+
+    context "when the display manager's service is disabled" do
+      it "returns false" do
+        expect(subject.enabled?).to eq(false)
+      end
+    end
+
+    context "when the display manager's service is enabled" do
+      let(:service_enabled) { true }
+
+      it "returns true" do
+        expect(subject.enabled?).to eq(true)
+      end
+    end
+  end
+
+  describe ".remote_access?" do
+    let(:remote_access_path) { ".sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS" }
+    let(:remote_access_value) { "yes" }
+    let(:service_enabled) { true }
+
+    before do
+      allow(Yast).to receive(:path).with(remote_access_path)
```

Comment:
np: Do you really need to mock this method? You could do something like:

```ruby
allow(Yast::SCR).to receive(:Read).with(Yast::Path.new(remote_access_path))
  .and_return(remote_access_value)
```

And get rid of the `:path` mock.